### ISSUE TITLE
RFE: Automatic Victory Condition parameter settings (When eligible)

### ISSUE THREAD

**Issue by Immortal-Fenix** [MegaMek/mekhq#4265]
### Environment
All versions of MegaMek could benefit from this change, in my opinion.

### Description
Missions that have simple goals, for example, "eliminate X% of enemy forces" or "Force Y% of enemy force to withdraw" could easily be automated into the megamek system from MekHQ, and would prevent accidental cheesing or forgetfulness on behalf of the player. More advanced mission objectives, like Recon or Breakthrough, would obviously still have to be tracked by the player. But I think it would be a pleasant change to have the game automatically update on your standard kill mission.

**Comment by SuperStucco** [comment]
MegaMek has victory conditions that can be set.  The problem is, for players such as myself the game terminates when those conditions are met.  The way I play, combat doesn't stop simply because some condition is met; instead one side or the other starts to pull back and retreat until one side is clearly in control of the field.  This results in opportunities for salvage, prisoners, and potential loss of valuable equipment and experienced pilots on both sides.  I'd wager I'm not alone in this, so there would need to be some means of turning this off or at least allowing play beyond what those conditions are set for.

**Comment by cooshoo** [comment]
I agree with @SuperStucco on this. The idea behind RFE is still good though. It would be nice if we could get a notification when conditions are met, either in chat, or maybe in the report?
